docs/FunctionalTestBuildScripts.md

```diff
@@ -85,6 +85,34 @@ A simple hello world program, compiled with `gcc 9.4.0` that generates an execut
 A simple hello world program, compiled with `gcc 9.3.0` that generates a .o object file. Script to reproduce:  
 `gcc-9 -Wall -c helloc.c -O2 -g -gdwarf-3 -o gcc.object_file.dwarf.3.o`
 
+## Managed_x64_VS2022_NetCore6_CSharp_HighEntropyVA_[True,False].dll
```

Comment:
>Managed_x64_VS2022_NetCore6_CSharp_HighEntropyVA_ [](http://example.com/codeflow?start=3&length=49)

Take a look at changes I made in the rule. See if you agree.

For this test file, our code says that any dotnet core entry point dll which is loaded into a 64-bit process will be forced into high entropy va. Which makes this setting effectively meaningless for that scenario.

Now, if users can set it reliably, we would recommend it.

Thoughts?

Reply:
1. I see in the if statement the `Requires32Bit` and `Prefers32Bit` is removed, agree. My understanding per the test file I created, even a file with both true can be running in 64 by default, so there is no point checking them.
2. We also removed `!portableExecutable.IsManaged`, I think it is by mistake?
3. We now have a second check `if (!coffHeader.Characteristics.HasFlag(Characteristics.LargeAddressAware))` and return not applicable, I think we don't need it because we have that case in the `Analyze()`, and in this case currently we ask people to enable it together with `HIGHENTROPYVA`.
4.  I think I may mistake your question here but let me first explain my thoughts: I see we have a comment in the code "A dotnet core entry point dll is itself loaded within a process that will always be configured for high entropy va, if available."
I am wondering if this is true and have a source of it, because the 2 test files I created, currently will get caught by this check right under this comment, and directly return not applicable.
However if I delete this comment with the check, my 2 test files can get analyzed and pass and fail, according to the setting I set in `<HighEntropyVA>True/False</HighEntropyVA>`.
So I brought this question for discussion.
If my assumption is wrong, my current two test files should be deleted because they will block the PR checks with current code.

Reply:
Thanks for fixing 2 and 3 and explaining the 4 does not matter what `HighEntropyVA` user set because will be taken care of by runtime.

Reply:
I have fixed the test and the dotnet format.
The current test does not work in Linux is not related to this PR, happening in main as well, I am looking to that.
I have approved the pr, ready for merge.